src/components/Transaction.js

```diff
@@ -0,0 +1,26 @@
+import React, {useState, useEffect} from 'react';
+
+export const Transaction = ({transaction}) => {
+    const [transactions, setTransactions] = useState(
+        JSON.parse(localStorage.getItem('transactions')) || []
+    );
+
+    // useEffect(() => {
+    //     localStorage.setItem('transactions', JSON.stringify(transactions))
+    // }, [transactions]);
+
+    const deleteTransaction = (id) => {
+        setTransactions(transactions.filter((transaction) => transaction.id !== id))
+        // TODO figure out why transaction isn't removed
+        //localStorage.setItem('transactions', JSON.stringify(transactions))
+    };
+
+    const sign = transaction.amount < 0 ? '-' : '+';
+
+    return (
+        <li className={transaction.amount < 0 ? 'minus' : 'plus'}>
+            {transaction.text} <span>{sign}{Math.abs(transaction.amount)}</span>
+            <button className="delete-btn" onClick={() => deleteTransaction(transaction.id)}>x</button>
```

Comment:
This doesn't work for some reason. I thought that I need to update local storage in useEffect (commented part), but it didn't work.

Reply:
Actually, it does work, but not in a way you expect it to work. The list is not updated because it does not know anything about the changes (see [this](https://github.com/Sophia440/kp-expense-tracker/pull/6/files#r750437918) comment), but the deletion does not work properly because you have N independent states (depending on the number of transactions) and on each deletion, you set different arrays to local storage.